src/rust/engine/options/src/args.rs

```diff
@@ -105,28 +150,40 @@ impl Args {
 
 impl OptionsSource for Args {
     fn display(&self, id: &OptionId) -> String {
-        Self::arg_name(id, Negate::False)
+        format!(
+            "--{}{}",
+            match &id.scope {
+                Scope::Global => "".to_string(),
+                Scope::Scope(scope) => format!("{}-", scope.to_ascii_lowercase()),
+            },
+            id.name("-", NameTransform::ToLower)
+        )
     }
 
     fn get_string(&self, id: &OptionId) -> Result<Option<String>, String> {
-        match self.find_flag(Self::arg_names(id, Negate::False))? {
-            Some((name, value, _)) => expand(value).map_err(|e| e.render(name)),
-            _ => Ok(None),
+        for arg in self.args.iter().rev() {
```

Comment:
It isn't quite obvious why we're iterating in reverse in many of these functions?

Reply:
I'll comment in the code, but it's because we want the last arg to win, for scalars and dicts (for lists we must iterate over all args as they may append instead of replacing).

Reply:
I have the impression the `--some-dict='+{"new_key": "..."}'` syntax means the append logic potentially also applies to dicts? (Going off https://www.pantsbuild.org/2.20/docs/using-pants/key-concepts/options#addreplace-semantics)

Or maybe this is something else?

Reply:
Ah yes, that didn't work even before this change, but should! Will fix.

Reply:
I will fix this bit in a followup change, since it didn't work before either, and it requires some interface changes that are out of scope for this PR.